functions/src/analytics.ts

```diff
@@ -58,20 +61,33 @@ const getUsersWithVisits = async (venueIdsArray: string[]) => {
 };
 
 const generateAnalytics: HttpsFunctionHandler<{
-  venueIds: string[] | string | undefined;
+  worldSlug: string | undefined;
 }> = async (data, context) => {
   assertValidAuth(context);
 
-  if (!context.auth) return;
+  if (!context.auth) {
+    throw new HttpsError("internal", `No authentication context`);
+  }
 
   await checkIsAdmin(context.auth.token.user_id);
 
-  const venueIdsArray = Array.isArray(data?.venueIds)
-    ? data.venueIds
-    : (data?.venueIds ?? "").split(",");
+  const matchingVenues = await admin
+    .firestore()
+    .collection("venues")
+    .where("worldId", "==", data.worldSlug)
+    .get();
+
+  if (matchingVenues.empty) {
+    throw new HttpsError(
+      "internal",
+      `The world ${data.worldSlug} does not have any venues`
+    );
+  }
+
+  const venueIds = matchingVenues.docs.map((venue) => venue.id);
```

Comment:
```suggestion
  const spaceIds = matchingVenues.docs.map((venue) => venue.id);
```

Reply:
Renamed in https://github.com/sparkletown/sparkle/pull/2826/commits/0633e23d3105f387087ff095df686fb6e7ce6139